src/settingsview.cpp

```diff
@@ -102,23 +119,21 @@ void SettingsView::browseMonitorDir()
     if (dir == monitorDir || dir.isEmpty()) {
         return;
     }
-    if (confirmDirDialog(dir, gt("monitor-dir-dialog-msg"), gt("monitor-dir-dialog-title"))) {
+    if (confirmDialogMonitorDir(dir, false)) {
         KiwixApp::instance()->getSettingsManager()->setMonitorDir(dir);
         ui->monitorClear->show();
         KiwixApp::instance()->getLibrary()->loadMonitorDir(dir);
         KiwixApp::instance()->getWatcher()->addPath(dir);
```

Comment:
Shouldn't you `KiwixApp::instance()->getWatcher()->removePath(previousDir);` here too (when `previousDir` is not empty)?